cms/djangoapps/contentstore/views/entrance_exam.py

```diff
@@ -0,0 +1,206 @@
+from milestones import api as milestones_api
+from models.settings.course_metadata import CourseMetadata
+from opaque_keys.edx.keys import CourseKey, UsageKey
+from opaque_keys import InvalidKeyError
+from student.auth import has_course_author_access
+from xmodule.modulestore.django import modulestore
+from xmodule.modulestore.exceptions import ItemNotFoundError
+from django.conf import settings
+
+__all__ = ['entrance_exam', ]
+
+
+@login_required
+@ensure_csrf_cookie
+def entrance_exam(request, course_key_string):
+    """
+    The restful handler for entrance exams.
+    It allows retrieval of all the assets (as an HTML page), as well as uploading new assets,
+    deleting assets, and changing the "locked" state of an asset.
+
+    GET
+        Retrieves the entrance exam module (metadata) for the specified course
+    POST
+        Adds an entrance exam module to the specified course.
+    DELETE
+        Removes the entrance exam from the course
+    """
+    course_key = CourseKey.from_string(course_key_string)
+
+    # Deny access if the user is valid, but they lack the proper object access privileges
+    if not has_course_author_access(request.user, course_key):
+        return HttpResponse(status=403)
+
+    # Retrieve the entrance exam module for the specified course (returns 404 if none found)
+    if request.method == 'GET':
+        return _get_entrance_exam(request, course_key)
+
+    # Create a new entrance exam for the specified course (returns 201 if created)
+    elif request.method == 'POST':
+        response_format = request.REQUEST.get('format', 'html')
+        http_accept = request.META.get('http_accept')
+        if response_format == 'json' or 'application/json' in http_accept:
+            ee_min_score = request.POST.get('entrance_exam_minimum_score_pct', None)
+
+            # if request contains empty value or none then save the default one.
+            entrance_exam_minimum_score_pct = float(settings.ENTRANCE_EXAM_MIN_SCORE_PCT)
+            if ee_min_score != '' and ee_min_score is not None:
+                entrance_exam_minimum_score_pct = float(ee_min_score)
+            return create_entrance_exam(request, course_key, entrance_exam_minimum_score_pct)
+        return HttpResponse(status=400)
+
+    # Remove the entrance exam module for the specified course (returns 204 regardless of existence)
+    elif request.method == 'DELETE':
+        return delete_entrance_exam(request, course_key)
+
+    # No other HTTP verbs/methods are supported at this time
+    else:
+        return HttpResponse(status=405)
+
+
+def create_entrance_exam(request, course_key, entrance_exam_minimum_score_pct):
+    """
+    api method to create an entrance exam.
+    First clean out any old entrance exams.
+    """
+    _delete_entrance_exam(request, course_key)
+    return _create_entrance_exam(request=request, course_key=course_key, entrance_exam_minimum_score_pct=entrance_exam_minimum_score_pct)
```

Comment:
nitpick: this is a bit long line. Except for entrance_exam_minimum_score_pct, you don't need calling parameter names for the first two.